scripts/run-eslint.ts

```diff
@@ -79,61 +152,48 @@ async function main() {
  *
  * @returns The parsed arguments.
  */
-function parseCommandLineArguments() {
-  return yargs(process.argv.slice(2))
+function parseCommandLineArguments(): CommandLineArguments {
+  const { cache, fix, quiet, ...rest } = yargs(process.argv.slice(2))
     .option('cache', {
       type: 'boolean',
       description: 'Cache results to speed up future runs',
       default: false,
     })
     .option('fix', {
       type: 'boolean',
-      description: 'Automatically fix problems',
+      description:
+        'Automatically fix all problems; pair with --quiet to only fix errors',
       default: false,
     })
     .option('quiet', {
       type: 'boolean',
-      description:
-        'Only report errors, disabling the warnings quality gate in the process',
+      description: 'Only report or fix errors',
       default: false,
     })
-    .help().argv;
+    .help()
+    .string('_').argv;
+
+  // Type assertion: The types for `yargs`'s `string` method are wrong.
+  const files = rest._ as string[];
+
+  return { cache, fix, quiet, files };
 }
 
 /**
- * Runs ESLint on the project files.
+ * Uses the given results to print the output that `eslint` usually generates.
  *
  * @param eslint - The ESLint instance.
- * @param options - The options for running ESLint.
- * @param options.quiet - Whether to only report errors (true) or not (false).
- * @param options.fix - Whether to automatically fix problems (true) or not
- * (false).
- * @returns A promise that resolves to the lint results.
+ * @param results - The results from running `eslint`.
  */
-async function runESLint(
```

Comment:
The diff is a bit rough here. Essentially I folded the contents of the `runESLint` function back into `main` and then extracted `printResults`. I felt this slight refactor was necessary because it was easier to see where we are using the output from `lintFiles` (a list of files and lint rules that were checked, which have been filtered to the given set of files) and where we are using a version of that list that is further filtered to just include errors.